src/App/Fossa/FossaAPIV1.hs

```diff
@@ -147,8 +148,12 @@ uploadAnalysis rootDir apiOpts ProjectRevision{..} metadata projects = fossaReq
           <> mkMetadataOpts metadata projectName
           -- Don't include branch if it doesn't exist, core may not handle empty string properly.
           <> maybe mempty ("branch" =:) projectBranch
-  resp <- req POST (uploadUrl baseUrl) (ReqBodyJson sourceUnits) jsonResponse (baseOpts <> opts)
-  pure (responseBody resp)
+  -- We do more filtering above and therefore must recheck for empty upload body
+  if null filteredProjects
+    then fatalText "No projects to upload (analyzed projects were filtered out by directory name)"
```

Comment:
Can we output the list of filtered projects here? This way users will know if they should instruct the cli to not filter them.

Reply:
They can't instruct the CLI not to filter them.  This filtering is based on the projects being in a `docs/` directory, for example.  This is completely unrelated to build-target filtering.  We don't currently support checking those folders from outside, and this wouldn't be the PR to address that.

I can log the relative paths to debug, and change this message to point to debug logs.